demos/common/cpp/models/src/classification_model.cpp

```diff
@@ -158,19 +158,16 @@ void ClassificationModel::prepareInputsOutputs(std::shared_ptr<ov::Model>& model
     model = ppp.build();
 
     // --------------------------- Adding softmax and topK output  ---------------------------
-    auto nodes = model->get_ops();
-    auto softmaxNodeIt = std::find_if(std::begin(nodes), std::end(nodes), [](const std::shared_ptr<ov::Node>& op) {
-        return std::string(op->get_type_name()) == "Softmax";
-    });
-
-    std::shared_ptr<ov::Node> softmaxNode;
-    if (softmaxNodeIt == nodes.end()) {
-        auto logitsNode = model->get_output_op(0)->input(0).get_source_output().get_node();
-        softmaxNode = std::make_shared<ov::op::v1::Softmax>(logitsNode->output(0), 1);
-    } else {
-        softmaxNode = *softmaxNodeIt;
+
+    if (model->get_output_size() != 1) {
+        throw std::logic_error("Expected model's number of output equal to 1 (but found: " + std::to_string(model->get_output_size()) + ")");
     }
+
+    auto logitsNode = model->get_output_op(0)->input(0).get_source_output().get_node();
```

Comment:
There are models which already have SoftMax, which names I don't remember. So this already may be SoftMax.

Reply:
Throughout 62 models from the supported [list](https://github.com/openvinotoolkit/open_model_zoo/blob/master/demos/classification_benchmark_demo/cpp/models.lst) there are 62 models with last layer of type "Result". They all work well with the change.